librosa/feature/inverse.py

```diff
@@ -195,26 +196,44 @@ def mfcc_to_mel(mfcc, n_mels=128, dct_type=2, norm='ortho', ref=1.0):
     ref : number or callable
         Reference power for (inverse) decibel calculation
 
+    lifter : number >= 0
+        If `lifter>0`, apply inverse liftering (inverse cepstral filtering):
+        `M[n, :] <- M[n, :] / (1 + sin(pi * (n + 1) / lifter)) * lifter / 2`
 
     Returns
     -------
     M : np.ndarray [shape=(n_mels, n)]
         An approximate Mel power spectrum recovered from `mfcc`
 
+    Warns
+    --------
+    RuntimeWarning
+        overflow encountered in power or
+        divide by zero encountered during inverse liftering.
 
     See Also
     --------
     mfcc
     melspectrogram
     scipy.fftpack.dct
     '''
+    if lifter > 0:
+        n_mfcc = mfcc.shape[0]
+        idx = np.arange(1, 1 + n_mfcc, dtype=mfcc.dtype)
+        lifter_sine = 1 + (lifter / 2) * np.sin(np.pi * idx / lifter)[:, np.newaxis]
+        lifter_sine = np.where(lifter_sine == 0, np.finfo(float).eps, lifter_sine)
+        mfcc = np.where(mfcc == 0, np.finfo(float).eps, mfcc) 
+        liftered_mfcc = mfcc / lifter_sine
+        mfcc = liftered_mfcc
```

Comment:
This part has a few issues that need to be fixed before we can merge.

1. Line 224: lifter_sine has exact zeros replaced by float epsilons.  Two things here: we'll almost never get exact zeros, and if you are going to replace these by machine epsilons, it should match the data type of `lifter_sine`, not be the default `float`.  You can get this in short-hand by calling `librosa.util.tiny(lifter_sine)`, but I'm not convinced this is a safe operation in general.
2. Why the in-filling of `mfcc` at line 225?  Dividing `0/lifter_sine` ought to be safe, but in general the `0/0` problem is undefined here, but I think any infilling should only be applied if we have a lifter that could zero out values.  If `lifter_sine` is always sufficiently positive, then we can safely infer that `mfcc==0` implies that the 0 value came pre-liftering, so post-liftering should stay 0 rather than becoming `eps/eps ~= 1`.  The same critique as above applies though, exact checks for 0 should be avoided.

Overall, I'm wary of trying too hard to prevent errors here.  Instead, I think it's safer to do the following:

1. Build the lifter array.
2. Check the lifter array for values likely to invoke underflow, and warn if they occur.
3. `mfcc = mfcc / lifter_sine` without any additional pre- or post-processing of `mfcc`.